pkg/service/service.go

```diff
@@ -843,13 +852,29 @@ func (s *Service) restoreServicesLocked() error {
 			// was deleted while cilium-agent was down).
 			restoredFromDatapath: true,
 		}
+		newSVC.maglev = newSVC.useMaglev()
 
 		for j, backend := range svc.Backends {
 			hash := backend.L3n4Addr.Hash()
 			s.backendRefCount.Add(hash)
 			newSVC.backendByHash[hash] = &svc.Backends[j]
 		}
 
+		// Recalculate Maglev lookup tables if the maps were removed due to
+		// the changed M param.
+		ipv6 := newSVC.frontend.IsIPv6()
+		if option.Config.DatapathMode == datapathOpt.DatapathModeLBOnly &&
+			newSVC.maglev && s.lbmap.IsMaglevLookupTableRecreated(ipv6) {
```

Comment:
For example, you can just replace the usage of this boolean with the call on L855